lib/active_merchant/billing/gateways/banwire.rb

```diff
@@ -32,61 +54,117 @@ def add_response_type(post)
         post[:response_format] = "JSON"
       end
 
-      def add_customer_data(post, options)
-        post[:user] = @options[:login]
-        post[:phone] = options[:billing_address][:phone]
-        post[:mail] = options[:email]
+      def add_auth_data(post, options)
+        post[:user] = @options[:login] if @options[:login] #- User (required) [alphanumeric] (Test: userdemo)
       end
 
-      def add_order_data(post, options)
-        post[:reference] = options[:order_id]
-        post[:concept] = options[:description]
+      def add_order_data(post, money, options)
+        post[:ORD_CURR_CD] = (options[:currency] || currency(money)).upcase #- Currency (optional) [text - length: 3] (Default: MXN)
+        post[:ORD_ID] = options[:order_id] #- Unique transaction ID (required) [alphanumeric - length: 1-22] (Example: 12345ABC)
+        post[:ORD_AMT] = amount(money) #- Transaction amount (required) [float - length: 10,2] (Example: 100.00)
+        post[:ORD_CONCEPT] = options[:description] #- Concept (required) [alphanumeric - lenght: 1-33] (Example: Compra de articulos)
```

Comment:
`:description` shouldn't be required; if the API requires it, please provide a default in case it's not supplied in `options`.